src/CtrlS/RoundState.java

```diff
@@ -35,23 +35,37 @@ public RoundState(GameState prevState, GameState currState) {
     }
 
     private int calculateCoin() {
+        int coin = 0;
+
+
+
 
         int baseCoin = roundScore / 10;
+        //레벨 보너스(levelBonus)
         int levelBonus = baseCoin * currState.getLevel();
-        int coin = baseCoin + levelBonus;
+        coin += levelBonus;
+
+        //정확도 보너스(accuracyBonus)
+        int accuracyBonus = 0;
 
         if (roundHitRate > 0.9) {
+            accuracyBonus += (int) (coin * 0.3);
             coin += (int) (coin * 0.3); // 30% 보너스 지급
             Core.getLogger().info("hitRate bonus occurs (30%).");
         } else if (roundHitRate > 0.8) {
+            accuracyBonus += (int) (coin * 0.3);
             coin += (int) (coin * 0.2); // 20% 보너스 지급
             Core.getLogger().info("hitRate bonus occurs (20%).");
+        } else if (roundHitRate > 0.7) {
+            accuracyBonus += (int) (coin * 0.3);
+            coin += (int) (coin * 0.1); // 10% 보너스 지급
+            Core.getLogger().info("hitRate bonus occurs (10%).");
         }
```

Comment:
현재 코드는 레벨 보너스가 포함된 coin에서 퍼센트를 계산하는 문제가 아직 있는 것 같습니다.
그리고 accuracyBonus는 계산만 되고 사용이 되지 않는 것 같습니다.

아래와 같이 accuracyBonus를 baseCoin을 기반으로 계산한 뒤, 조건문 밖에서 coin에 accuracyBonus를 더하는 게 어떨까요?
```suggestion
        if (roundHitRate > 0.9) {
            accuracyBonus += (int) (baseCoin * 0.3);
            Core.getLogger().info("hitRate bonus occurs (30%).");
        } else if (roundHitRate > 0.8) {
            accuracyBonus += (int) (baseCoin * 0.2);
            Core.getLogger().info("hitRate bonus occurs (20%).");
        } else if (roundHitRate > 0.7) {
            accuracyBonus += (int) (baseCoin * 0.1);
            Core.getLogger().info("hitRate bonus occurs (10%).");
        }
        coin += accuracyBonus
```

Reply:
> 현재 코드는 레벨 보너스가 포함된 coin에서 퍼센트를 계산하는 문제가 아직 있는 것 같습니다. 그리고 accuracyBonus는 계산만 되고 사용이 되지 않는 것 같습니다.
> 
> 아래와 같이 accuracyBonus를 baseCoin을 기반으로 계산한 뒤, 조건문 밖에서 coin에 accuracyBonus를 더하는 게 어떨까요?

동의 합니다. 한 보너스가 다른 보너스 계산에 개입이 되서는 안되고 보너스 계산은 오로지 baseCoin으로만 이루어져 따로 변수에 저장되고 baseCoin에 각 보너스가 더해져 최종 해당 라운드에서의 획득 코인이 되어야 합니다.

Reply:
확인했습니다. 제안 사항을 커밋에 반영했습니다.